src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -272,22 +287,35 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 				newExpr = memberAccess.Expression;
 				return Transform( newExpr );
 			}
+		} else if( memberAccess is not null && ShouldWrapMemberAccessesInTaskRun( memberAccess ) ) {
+			m_disableTaskRunWarningFlag = true;
+			return SyntaxFactory.ParseExpression( $"Task.Run(() => {invocationExpr}).Result" );
 		}
+
 		return invocationExpr
 			.WithExpression( Transform( invocationExpr.Expression ) )
 			.WithArgumentList( TransformAll( invocationExpr.ArgumentList, Transform ) );
 	}
 
-	bool ShouldRemoveReturnedMemberAccess( MemberAccessExpressionSyntax memberAccessExpr )
-	  => memberAccessExpr.Name.Identifier.ValueText switch {
-		  "FromResult" => true,
-		  "CompletedTask" => true,
-		  _ => false
-	  };
+	bool ShouldRemoveReturnedMemberAccesses( MemberAccessExpressionSyntax memberAccessExpr ) {
+		return ( memberAccessExpr.Expression.ToString(), memberAccessExpr.Name.Identifier.ValueText ) switch
+		{
+			( "Task", "FromResult" ) => true,
+			( "Task", "CompletedTask" ) => true,
+			_ => false
+		};
+	}

```

Comment:
This change appears ~unrelated?

Reply:
I think it deserves its own PR because we're using Squash and Merge which is going to put all our commits into one once we merge, and when you look at a `git blame` later, it will be confusing to see this change being attributed to _Add Generator to wrap some async calls in Task.Run().Result_. It deserves its own explanation :smile: .

Reply:
Kk, I moved the changes from here to https://github.com/Brightspace/D2L.CodeStyle/pull/897 so we can merge that in afterwards